server/viame_server/viame.py

```diff
@@ -126,74 +122,91 @@ def validate_files(self, files):
+                    auxiliary["_id"],
+                    girder_job_title=(
+                        "Converting {} to a web friendly format".format(
+                            str(item["_id"])
+                        )
+                    ),
+                    girder_client_token=str(token["_id"]),
+                )
 
-        for item in videoItems:
-            convert_video.delay(
-                GetPathFromItemId(str(item["_id"])),
-                str(item["folderId"]),
-                str(upload_token["_id"]),
-                auxiliary["_id"],
-                girder_job_title=(
-                    "Converting {} to a web friendly format".format(str(item["_id"]))
-                ),
+            # transcode IMAGERY if necessary
+            imageItems = Folder().childItems(
+                folder, filters={"lowerName": {"$regex": imageRegex}}
+            )
+            safeImageItems = Folder().childItems(
+                folder, filters={"lowerName": {"$regex": safeImageRegex}}
             )
 
-        # transcode IMAGERY if necessary
-        imageItems = Folder().childItems(
-            folder, filters={"lowerName": {"$regex": imageRegex}}
-        )
-        safeImageItems = Folder().childItems(
-            folder, filters={"lowerName": {"$regex": safeImageRegex}}
-        )
+            if imageItems.count() > safeImageItems.count():
+                convert_images.delay(
+                    folder["_id"],
+                    girder_client_token=str(token["_id"]),
+                    girder_job_title=f"Converting {folder['_id']} to a web friendly format",
+                )
+            elif imageItems.count() > 0:
+                folder["meta"]["annotate"] = True
 
-        if imageItems.count() > safeImageItems.count():
-            convert_images.delay(
-                folder["_id"],
-                girder_client_token=str(upload_token["_id"]),
-                girder_job_title=f"Converting {folder['_id']} to a web friendly format",
+            # transform KPF if necessary
+            ymlItems = Folder().childItems(
+                folder, filters={"lowerName": {"$regex": ymlRegex}}
             )
-        elif imageItems.count() > 0:
-            folder["meta"]["annotate"] = True
+            if ymlItems.count() > 0:
+                # There might be up to 3 yamls
+                allFiles = [Item().childFiles(item)[0] for item in ymlItems]
+                saveTracks(folder, meva_serializer.load_kpf_as_tracks(allFiles), user)
+                ymlItems.rewind()
+                for item in ymlItems:
+                    Item().move(item, auxiliary)
 
-        # Preprocess CSV if necessasry
+            Folder().save(folder)
+
+        # transform CSV if necessasry
         csvItems = Folder().childItems(
-            folder, filters={"lowerName": {"$regex": csvRegex}}
+            folder,
+            filters={"lowerName": {"$regex": csvRegex}},
+            sort=[("created", pymongo.DESCENDING,)],
```

Comment:
A small thing mainly for me. Is Girder built directly off of MongoDb/could you grab this from girder? Or is it just easier/clearer this way?

Reply:
Girder models are built on pymongo.  These functions are just sort of thin wrappers around pymongo queries, and they support adding additional pymongo query params.

Reply:
https://api.mongodb.com/python/current/api/pymongo/collection.html#pymongo.collection.Collection.find